extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/TranslatableInvalidHostConnectionException.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.restrict;
+
+import org.apache.guacamole.language.TranslatableGuacamoleSecurityException;
+import org.apache.guacamole.language.TranslatableMessage;
+
+/**
+ * An exception that represents an invalid login or connection due to
+ * restrictions based on the host from which the action should be allowed.
+ */
+public class TranslatableInvalidHostConnectionException 
+        extends TranslatableGuacamoleSecurityException {
+
+    /**
+     * The serial version ID of this class.
+     */
+    private static final long serialVersionUID = 1L;
+    
+    /**
+     * Create a new host-based connection exception with the given message and
+     * translation string that can be processed by Guacamole's translation
+     * service.
+     * 
+     * @param message
+     *     The non-translatable, human-readable message containing details
+     *     of the exception.
+     * 
+     * @param translatableMessage 
+     *     The translation key for this exception that can be processed through
```

Comment:
This isn't necessarily just a key - consider rewording this param description.

Reply:
Fixed via rebase.